### PR TITLE
docs: add docs for using rule tester with node:test

### LINKED ISSUE #8273: Docs: Add RuleTester docs for working with node:test
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

@justinfagnani [posted in our Discord](https://discord.com/channels/1026804805894672454/1197313042875154542) some questions about `RuleTester` errors around `describe` and `afterAll`. The thread ended up noting that we don't have a docs section for `node:test` yet - just Jest and Vitest. Now that `node:test` is stable it'd be good to add a docs section for it. Accepting PRs!

Sibling issue to #8274, which suggests a general/other section.

### Affected URL(s)

https://typescript-eslint.io/packages/rule-tester/#with-specific-frameworks

### PR DESCRIPTION
**Pull request by NotWoods**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8273
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Adds an example of RuleTester with the node:test runner, which uses Node's CLI flags to import the setup file (as there's no equivalent test runner specific flag). The test file will get imported for every test, but that should be fine since we just copy some variables.

### CODE REVIEW

File: docs/packages/Rule_Tester.mdx [hunk 1]
```diff
@@ -233,6 +233,28 @@ RuleTester.itOnly = vitest.it.only;
...
+```ts
+// setup.js
+import * as test from 'node:test';
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+RuleTester.afterAll = test.afterAll;
+
+RuleTester.it = test.it;
+RuleTester.itOnly = test.it.only;
+RuleTester.describe = test.describe;
```
Comment (JoshuaKGoldberg) [inline docs/packages/Rule_Tester.mdx:249]: Nit: any reason for the blank line?

```suggestion
RuleTester.describe = test.describe;
RuleTester.it = test.it;
RuleTester.itOnly = test.it.only;
```

**Review by JoshuaKGoldberg — DISMISSED** [review]
LGTM, thanks! 🙌

File: docs/packages/Rule_Tester.mdx (hunk 1 above)
Comment (NotWoods) [reply docs/packages/Rule_Tester.mdx:249]: I was trying to mirror the above example (which admittedly has a comment too)

**Review by JoshuaKGoldberg — APPROVED** [review]
🚀

**Review by auvred — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**